addons/zeus/functions/fnc_moduleGarrison.sqf

```diff
@@ -0,0 +1,57 @@
+/*
+ * Author: alganthe
+ * Module calling the garrison function
+ *
+ * Arguments:
+ * 0: The module logic <OBJECT>
+ * 1: Position of the module <POSITION>
+ * 2: Radius of the task <NUMBER>
+ * 3: Filling mode of the garrison function <NUMBER>
+ * 4: Enable or not top down filling <BOOLEAN>
+ *
+ * Return Value:
+ * None
+*/
+
+#include "script_component.hpp"
+
+params ["_logic", "_pos", "_radius" ,"_mode" , "_topDownMode"];
+
+private _unit = (attachedTo _logic);
+private _building = nearestBuilding (getPosASL _unit);
+
+// Handles errors
```

Comment:
Validation of the module target should only be taking place in the ui function (it doesn't make sense to show zeus a UI and then tell them they placed the module wrong only after they close it).